### PR TITLE
feat(SDRS): add a new resource sdrs replication attach

### PR DESCRIPTION
**Pull request by deer-hang**
**What this PR does / why we need it**:
add a new resource sdrs replication attach
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

- This PR depends on #3117, #3119 and #3121
- Make sure the PR #3117, #3119 and #3121 have been merged before merge this PR.

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/sdrs' TESTARGS='-run TestAccReplicationAttach_basic'
...
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/sdrs -v -run TestAccReplicationAttach_basic -timeout 360m -parallel 4 
=== RUN   TestAccReplicationAttach_basic 
=== PAUSE TestAccReplicationAttach_basic
=== CONT  TestAccReplicationAttach_basic
--- PASS: TestAccReplicationAttach_basic (427.11s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/sdrs      427.181s
```

### CODE REVIEW

File: docs/resources/sdrs_replication_attach.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
+---
+subcategory: "Storage Disaster Recovery Service (SDRS)"
+---
+
+# huaweicloud_sdrs_replication_attach
+
+Manages an SDRS replication attach resource within HuaweiCloud.
+
+-> **NOTE:** Please make sure the status of the protection group must be available or protected, the status of the
```
Comment (chengxiangdong) [inline docs/resources/sdrs_replication_attach.md]:
```suggestion
-> Please make sure the status of the protection group must be available or protected, the status of the
```

File: huaweicloud/services/sdrs/resource_huaweicloud_sdrs_replication_attach.go [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+			}
+		}
+		return diag.FromErr(err)
+	}
+
+	deleteTimeoutSec := int(d.Timeout(schema.TimeoutDelete).Seconds())
+	if err := attachreplication.WaitForJobSuccess(client, deleteTimeoutSec, n.JobID); err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
```
Comment (chengxiangdong) [inline huaweicloud/services/sdrs/resource_huaweicloud_sdrs_replication_attach.go:180]:
```suggestion
	d.SetId("")
	return nil
```

File: huaweicloud/services/sdrs/resource_huaweicloud_sdrs_replication_attach.go [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+}
+
+func formatAttachId(instanceID string, replicationID string) string {
+	return fmt.Sprintf("%s:%s", instanceID, replicationID)
+}
+
+func extractAttachId(resourceID string) (instanceID string, replicationID string, err error) {
```
Comment (chengxiangdong) [inline huaweicloud/services/sdrs/resource_huaweicloud_sdrs_replication_attach.go]: It is not recommended to specify variables when defining the return type.

**Review by chengxiangdong — COMMENTED** [review]
The line width is 150, and some functions do not need to wrap.

File: huaweicloud/services/sdrs/resource_huaweicloud_sdrs_replication_attach.go (hunk 2 above)
Comment (deer-hang) [reply huaweicloud/services/sdrs/resource_huaweicloud_sdrs_replication_attach.go:180]: The deletion method will automatically set the ID to empty internally.

File: huaweicloud/services/sdrs/resource_huaweicloud_sdrs_replication_attach.go (hunk 3 above)
Comment (deer-hang) [reply huaweicloud/services/sdrs/resource_huaweicloud_sdrs_replication_attach.go]: Defining variable names is to distinguish multiple return values of the same type.

**Comment by deer-hang** [comment]
```
 make testacc TEST='./huaweicloud/services/acceptance/sdrs' TESTARGS='-run TestAccReplicationAttach_basic'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/sdrs -v -run TestAccReplicationAttach_basic -timeout 360m -parallel 4 
=== RUN   TestAccReplicationAttach_basic 
=== PAUSE TestAccReplicationAttach_basic
=== CONT  TestAccReplicationAttach_basic
--- PASS: TestAccReplicationAttach_basic (421.28s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/sdrs      421.381s
```

File: huaweicloud/services/sdrs/resource_huaweicloud_sdrs_replication_attach.go [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+	client, err := cfg.SdrsV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating SDRS client: %s", err)
+	}
+
+	instanceID := d.Get("instance_id").(string)
+	replicationID := d.Get("replication_id").(string)
+	if err != nil {
+		return diag.FromErr(err)
+	}
```
Comment (ShiChangkuo) [inline huaweicloud/services/sdrs/resource_huaweicloud_sdrs_replication_attach.go]: these lines are useless, please check and delete them.

File: huaweicloud/services/sdrs/resource_huaweicloud_sdrs_replication_attach.go [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+
+	mErr := multierror.Append(
+		nil,
+		d.Set("region", region),
+		d.Set("device", attachment.Device),
+		d.Set("replication_id", attachment.Replication),
+		d.Set("status", n.Status),
+	)
+	return diag.FromErr(mErr.ErrorOrNil())
```
Comment (ShiChangkuo) [inline huaweicloud/services/sdrs/resource_huaweicloud_sdrs_replication_attach.go:134]: Please set the resource ID again to update the previous format.

File: huaweicloud/services/sdrs/resource_huaweicloud_sdrs_replication_attach.go (hunk 4 above)
Comment (deer-hang) [reply huaweicloud/services/sdrs/resource_huaweicloud_sdrs_replication_attach.go]: OK

**Pull request merged**